libvcx/src/credential_def.rs

```diff
@@ -357,29 +291,38 @@ pub fn create_and_publish_credentialdef(source_id: String,
                                         schema_id: String,
                                         tag: String,
                                         revocation_details: String) -> VcxResult<u32> {
-    trace!("create_new_credentialdef >>> source_id: {}, name: {}, issuer_did: {}, schema_id: {}, revocation_details: {}",
+    trace!("create_and_publish_credentialdef >>> source_id: {}, name: {}, issuer_did: {}, schema_id: {}, revocation_details: {}",
            source_id, name, issuer_did, schema_id, revocation_details);
 
     let revocation_details: RevocationDetails = _parse_revocation_details(&revocation_details)?;
 
-    // Creates Credential Definition and Revocation Definition in wallet
     let (cred_def_id, cred_def_json, rev_reg_id, rev_reg_def, rev_reg_entry) = _create_credentialdef(&issuer_did, &schema_id, &tag, &revocation_details)?;
 
-    // Publish Credential Definition on the ledger
-    let cred_def_payment_txn = anoncreds::publish_cred_def(&issuer_did, &cred_def_json)?;
+    let (rev_def_payment, rev_delta_payment, cred_def_payment_txn) = match _try_get_cred_def_from_ledger(&issuer_did, &cred_def_id) {
+        Ok(Some(ledger_cred_def_json)) => {
+            if serde_json::from_str::<serde_json::Value>(&cred_def_json).map_err(|err| VcxError::from_msg(VcxErrorKind::SerializationError, format!("Failed to deserialize generated cred def json: {}\nError: {}", cred_def_json, err)))? != 
+                serde_json::from_str::<serde_json::Value>(&ledger_cred_def_json).map_err(|err| VcxError::from_msg(VcxErrorKind::SerializationError, format!("Failed to deserialize cred def json from ledger: {}\nError: {}", ledger_cred_def_json, err)))? {
+                return Err(VcxError::from_msg(VcxErrorKind::CreateCredDef, "Desired credential definition differs from the one on the ledger"))
+            }
+            (None, None, None)
```

Comment:
Let's return error also here, simply if we find out that creddef already exists on ledger.